util/uuid.h

```diff
@@ -0,0 +1,150 @@
+  return lhs.data < rhs.data;
+}
+
+inline bool operator==(const RfcUuid &lhs, const RfcUuid &rhs) {
+  return lhs.data == rhs.data;
+}
+
+// A "mostly unique ID" encodable into 20 characters in base 36 ([0-9][A-Z])
+// This provides a compact, portable text representation with 103.4 bits of
+// entropy. This was originally developed for DB Session IDs (see
+// DB::GetDbSessionId(). All zeros means unset or empty.
+//
+// This is considered suitable for identifying medium-frequency events or
+// narrow scope events. For example, if 1 million computers each generate a
+// random RocksMuid every minute for a year, or one computer generates a
+// million every second for a week, the chance of a RocksMuid being repeated
+// in that time is 1 in millions. See Env::GenerateRawUuid() and
+// FromRawLoseData().
+//
+// In text form, the digits are in standard written order: lowest-order last,
+// except each piece (lower64 and upper64) only holds 10 base-36 digits.
+// Thus, each uint64_t piece is < kPieceMod. FIXME
+struct RocksMuid {
+  RawUuid scaled_data = 0;
+
+  // Parses from the 20-digit base-36 text form. Returns Corruption if
+  // malformed.
+  static Status Parse(const Slice &input, RocksMuid *out);
+
+  // This converts a RawUuid (also OK: RfcUuid) to a RocksMuid by
+  // FIXME on both 64-bit pieces. Assuming the input is random with
+  // 128 bits of entropy, about 24.6 bits of entropy are lost in the
+  // conversion.
+  static RocksMuid FromRawLoseData(const RawUuid &raw);
+
+  // For generating a human readable string, base-36 with 20 digits
+  std::string ToString() const;
+  // For generating a human readable string (base-36 with 20 digits)
+  // into an existing buffer. No trailing nul is written or expected.
+  // Returns a Slice for the buffer and 20 written digits.
+  Slice PutString(char *buf_of_20) const;
+
+ private:
+  static Unsigned128 ReducedToScaled(const Unsigned128 &reduced);
+  static void ScaledToPieces(const Unsigned128 &scaled, uint64_t *upper_piece,
+                             uint64_t *lower_piece);
+  static Unsigned128 PiecesToReduced(uint64_t upper_piece,
+                                     uint64_t lower_piece);
+  static Unsigned128 ScaledToReduced(const Unsigned128 &scaled);
+
+  // 36 to the 10th power
+  static constexpr uint64_t kPieceMod = 3656158440062976;
+};
+
+inline bool operator<(const RocksMuid &lhs, const RocksMuid &rhs) {
+  return lhs.scaled_data < rhs.scaled_data;
+}
+
+inline bool operator==(const RocksMuid &lhs, const RocksMuid &rhs) {
+  return lhs.scaled_data == rhs.scaled_data;
+}
+
+RawUuid AddInCounterA(const RawUuid &u, uint64_t counter);
+RawUuid AddInCounterB(const RawUuid &u, uint64_t counter);
+RawUuid AddInCounterC(const RawUuid &u, uint64_t counter);
+
+RawUuid SubtractOffCounterA(const RawUuid &u, uint64_t counter);
+RawUuid SubtractOffCounterB(const RawUuid &u, uint64_t counter);
+RawUuid SubtractOffCounterC(const RawUuid &u, uint64_t counter);
```

Comment:
I know these are private and in an internal, non-public class but I have no idea when I would use the CounterA vs CounterB/C methods.  Perhaps some comments somewhere?